src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,52 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_CAPACITY = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = new Object[MAX_CAPACITY];
+        values = new Object[MAX_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == 0) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            for (int i = 0; i < size && size <= MAX_CAPACITY; i++) {
+                if (key == null && keys[i] == null) {
+                    values[i] = value;
+                    return;
+                } else if (keys[i] != null && keys[i].equals(key)) {
+                    values[i] = value;
+                    return;
+                }
+            }
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && keys[i] == null) {
```

Comment:
```suggestion
            if (key == keys[i]) {
```
it is a nice way to compare nulls